src/ui/application-window.vala

```diff
@@ -251,6 +254,22 @@ namespace Peek.Ui {
       record_button.set_label (_("Record as %s").printf (format_name));
       pop_format.hide ();
     }
+    
+    async void update_time () {
+      double seconds;
+      Timer timer = new Timer ();
```

Comment:
What about using add_seconds_full? https://valadoc.org/glib-2.0/GLib.Timeout.add_seconds_full.html

Reply:
I initalially choose  not to use Timeout because
>Note that timeout functions may be delayed, due to the processing of other event sources. Thus they should not be relied on for precise timing

But I will give a test!

Reply:
That's true. You could still store the start time (in seconds) and on each interval calculate the elapsed time. That way a small time slip in the Timeout calls does not lead to wrongly displayed time.

Reply:
And yes, add_seconds_full could be tried. There is some info that the first call can be delayed by up to one second, though. Never tried it.